components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,836 @@
+                    var isValidElements = (validateRangeOfDay([startingDayOfMonthOptionArr[0]], 1, 31) &&
+                                                        validateRangeOfDay([startingDayOfMonthOptionArr[1]], 1, 31));
+                    var isValidFirstElem = (startingDayOfMonthOptionArr[0] === '*' &&
+                                                        validateRangeOfDay([startingDayOfMonthOptionArr[1]], 1, 31));
+                    return isValidElements || isValidFirstElem;
+                } else if(dayOfMonth.includes('-') && dayOfWeek === '?') {
+                    var dayOfMonthRangeArr = dayOfMonth.split('-');
+                    if(dayOfMonth.endsWith('-')){
+                        errorMsg = "Day of month values should not end with - ";
+                    } else {
+                        errorMsg = "Day of month values must be between 1 and 31";
+                    }
+                    return validateRangeOfDay(dayOfMonthRangeArr, 1, 31) &&
+                                    (parseInt(dayOfMonthRangeArr[0]) < parseInt(dayOfMonthRangeArr[1]));
+                } else {
+                    if(dayOfMonth === ""){
+                        errorMsg = "Day of month values should not end with , ";
+                    } else {
+                        errorMsg = "Day of month values must be between 1 and 31";
+                    }
+                    return validateRangeOfDay([dayOfMonth], 1, 31) ||
+                        (dayOfMonth === '*' && dayOfWeek !== '*') || (dayOfMonth === '?' && dayOfWeek !== '?') ||
+                        (dayOfWeek === '?' && (dayOfMonth === 'LW' || dayOfMonth === 'L'));
+                }
+            });
+        };
+
+        var monthArray = ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec'];
+
+        var validateStringValue = function(monthArr, dataArr) {
+            return monthArr.every(month => {
+                return dataArr.includes(month.toLowerCase());
+            })
+        };
+
+        var validateMonthValue = function(month) {
+            if(month.includes('_')){
+                errorMsg = "Invalid cron expression";
+                return false;
+            }
+            if (month.search(constants.REGEX_FOR_MONTH) !== -1) {
+                errorMsg = "Invalid cron expression";
+                return false;
+            }
+            var list = month.split(",");
+            return list.every(function (month){
+                if(month.includes('/')) {
+                    if(month.endsWith('/')){
+                        errorMsg = "Month values should not end with / ";
+                    } else {
+                        errorMsg = "Month values must be between 1 and 12";
+                    }
+                    var startingDayOfMonthOptionArr = month.split('/');
+                    var isValidElements = (validateRangeOfDay([startingDayOfMonthOptionArr[0]], 1, 12) &&
```

Comment:
validateRangeOfDay why do u pass 1, 12 what are we validating here ?

Reply:
this method is validating the values of the month(1-12) or dayOfweek(1-7) or dayOfMonth(1-31) are in the range. In here I pass 1,12 to validate the month range.

Reply:
ok